core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -197,4 +197,31 @@ private RelDataType struct(RelDataType...relDataTypes) {
     }
     return builder.build();
   }
+
+  @Test public void testEqualSansNullability() {
+    final RelDataTypeFactory factory = f.typeFactory;
+
+    RelDataType bigIntType = f.sqlBigInt;
+    RelDataType nullableBigIntType = f.sqlBigIntNullable;
+    RelDataType varCharType = f.sqlVarchar;
+
+    // different types
+    assertThat(
+        SqlTypeUtil.equalSansNullability(factory, bigIntType, varCharType), is(false));
+    assertThat(
+        SqlTypeUtil.equalSansNullability(bigIntType, varCharType), is(false));
```

Comment:
Please factor this into a method and make the assertion messages distinguishable.

Reply:
Done. Thanks for the good suggestion.